cmd/collector/app/collector.go

```diff
@@ -91,7 +102,7 @@ func (c *Collector) Start(builderOpts *CollectorOptions) error {
 		SamplingStore: c.strategyStore,
 		Logger:        c.logger,
 	}); err != nil {
-		c.logger.Fatal("could not start gRPC collector", zap.Error(err))
+		return newCollectorError("could not start gRPC collector", err)
```

Comment:
this seems an overkill, I would simply do `return fmt.Errorf("could not start gRPC collector: %w", err)`, which would properly wrap the error anyway.

Reply:
I didn't dare to do it. I think it's much better too.

Reply:
There is no need to worry about logger fields here, we just need to wrap the error with explaining messages using Errorf. This is how it's done everywhere else in this project when returning nested errors.

Reply:
More clear thank you !